paf-mvp-core-js/src/express/node.ts

```diff
@@ -166,23 +174,30 @@ export class Node implements INode {
         next(error);
         return;
       }
-      const validation = this.jsonValidator.validate(jsonSchema, decodedData);
-      if (!validation.isValid) {
-        const details = validation.errors.map((e) => e.message).join(' - ');
-        const error: NodeError = {
-          type: NodeErrorType.INVALID_QUERY_STRING,
-          details,
-        };
-        if (isRedirect) {
-          const redirectURL = buildErrorRedirectUrl(new URL(req.header('referer')), 400, error);
-          httpRedirect(res, redirectURL.toString());
+      try {
+        const validation = this.jsonValidator.validate(jsonSchema, decodedData);
+        if (!validation.isValid) {
+          const details = validation.errors.map((e) => e.message).join(' - ');
+          const error: NodeError = {
+            type: NodeErrorType.INVALID_QUERY_STRING,
+            details,
+          };
+          if (isRedirect) {
+            const redirectURL = buildErrorRedirectUrl(new URL(req.header('referer')), 400, error);
+            httpRedirect(res, redirectURL.toString());
+          } else {
+            res.status(400);
+            res.json(error);
+          }
+          next(error);
```

Comment:
Yes, GitHub does a bad job in this diff.
I added the call to the error handler to make sure it will be logged correctly. But anyway, the next step is to remove the block above:
```typescript
if (isRedirect) {
            const redirectURL = buildErrorRedirectUrl(new URL(req.header('referer')), 400, error);
            httpRedirect(res, redirectURL.toString());
          } else {
            res.status(400);
            res.json(error);
          }
```

This should be moved to the error handler.

Reply:
Ok it makes sense in this case.